### PR TITLE
Issue/13328 restore backup adapters

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13328 

This PR is Part III of a consolidation effort of the Jetpack backup/restore features. This part focuses on the the adapters. Changes are mostly in the import statements. For reference: Part I #13788 & Part II #13789

- Consolidated all adapters from backup and restore into `JetpackBackupRestoreAdapter`. Moved this class into the common folder. 

**Merge Instructions**
- Make sure #13788 and #13789 have been merged
- Remove Not ready for merge label
- Merge as normal

**Note:**
- I made one change in jetpack_backup_restore_fragment.xml to eliminate the unused id. Just caught my eye and I didn't want to forget to take care of it. 

**To test:**
- Run the app and make sure the backup/download still function as expected

PR submission checklist:
- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/common/adapters/JetpackBackupRestoreAdapter.kt [hunk 1]
```diff
@@ -34,8 +36,11 @@ class RestoreDetailsAdapter(
             ViewType.ICON.id -> JetpackIconViewHolder(imageManager, parent)
             ViewType.HEADER.id -> JetpackHeaderViewHolder(uiHelpers, parent)
             ViewType.DESCRIPTION.id -> JetpackDescriptionViewHolder(uiHelpers, parent)
+            ViewType.PROGRESS.id -> JetpackProgressViewHolder(uiHelpers, parent)
             ViewType.ACTION_BUTTON.id -> JetpackButtonViewHolder(uiHelpers, parent)
             ViewType.CHECKBOX.id -> JetpackCheckboxViewHolder(uiHelpers, parent)
+            ViewType.BACKUP_RESTORE_ADDITIONAL_INFORMATION.id ->
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/common/adapters/JetpackBackupRestoreAdapter.kt:42]: ❓ I noticed these two ViewTypes have "BACKUP_RESTORE" prefix, but the others don't have it. Should we remove it or is it intentional?

**Review by malinajirka — APPROVED** [review]
Thanks @zwarm! Great improvement, it's nice to see that you managed to remove so many lines of unnecessary code! Feel free to merge it when the parent PRs get merged.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/common/adapters/JetpackBackupRestoreAdapter.kt (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/common/adapters/JetpackBackupRestoreAdapter.kt:42]: @malinajirka  - Yes, it's intentional. We decided that anything exclusive to a feature should be prefixed. The same pattern in used by scan. Thanks

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/common/adapters/JetpackBackupRestoreAdapter.kt:42]: Could you please elaborate a bit more on this. I'm not sure I follow. `JetpackBackupRestoreAdapter` is used by two features (backup and restore)? `BACKUP_RESTORE_ADDITIONAL_INFORMATION` is also used by two features? Which part is exclusive to a feature?

**Pull request merged**